### ISSUE TITLE
What is the link to debug symbols server of .NET Core?

### ISSUE THREAD

**Issue by xperiandri** [dotnet/home#69]

**Comment by gkhanna79** [comment]
CC @markwilkie @dagood

**Comment by lt72** [comment]
Symbols for .Net Core 1.1. are available as a download from the Microsoft Download Center at [this URL](https://go.microsoft.com/fwlink/?LinkID=835699). This package includes both Linux and Windows symbol packages. 
You can also acquire the symbols directly from the [Microsoft Symbol Server](https://msdn.microsoft.com/en-us/library/windows/desktop/ee416588(v=vs.85).aspx#symbol_servers) is you are debugging on a .Net Core Windows application. Linux is not supported yet from the symbol server, but we may be able to enable that soon.

**Comment by dagood** [comment]
To get the Linux and Windows symbol packages for other releases, you can also find Microsoft Download Center links in the release notes in the dotnet/core repository:

 * [1.0.4.md#net-core-104-symbol-packages](https://github.com/dotnet/core/blob/master/release-notes/1.0/1.0.4.md#net-core-104-symbol-packages)
 * [1.1.md#net-core-11-symbol-packages](https://github.com/dotnet/core/blob/master/release-notes/1.1/1.1.md#net-core-11-symbol-packages)
 * [1.1.1.md#net-core-111-symbol-packages](https://github.com/dotnet/core/blob/master/release-notes/1.1/1.1.1.md#net-core-111-symbol-packages)

**Comment by xperiandri** [comment]
Must this work?
![image](https://cloud.githubusercontent.com/assets/2365592/26010944/a477047c-3757-11e7-9b36-ee426e7c3170.png)

**Comment by lt72** [comment]
yes, it should work if you use an official release, such as preview1:
https://www.microsoft.com/net/core/preview#windowscmd

**Comment by xperiandri** [comment]
I use 1.1
Is the first source equivalent to the second one in my above screenshot?

**Comment by lt72** [comment]
yes, it is

**Comment by xperiandri** [comment]
So what can be the issue where symbols for `Microsoft.Exnesions.*` and `Microsoft.AspNetCore.*` packages are not downloaded?

**Comment by lt72** [comment]
Both of those are packages from ASP.NET core, which were not uploaded to sym server using the same process for other packaages from the framework (e.g. CoreFX). We may have some missing symbols, I'll check...

**Comment by xperiandri** [comment]
Yes, please. Because it is very inconvenient that I can step through sources. I got an issue with very undescriptive error and couldn't have a look deeper that's happened.

**Comment by lt72** [comment]
@Eilon : were the symbols for ASP.Net Core for 1.1. uploaded to //symweb (msdl)?

**Comment by Eilon** [comment]
@lt72 - I think @vancem uploaded some ASP.NET Core 1.1 symbols somewhere, but I'm not sure. No one on my team uploaded them anywhere.

**Comment by vancem** [comment]
@Eilon.  I dealt with the Native PDBs (which are only needed by profilers, not the debugger).  For version 1.1 ASP.NET Dlls I only did 32 bit ones because from what I could tell version 1.1 does not support dotnet-store, and the only scenario that was generating Native code files for ASP.NET DLLs was Antares, which did it by hand (and you pointed me at them).  

However I believe all of this is irrelevant for the case above, because this is for the debugger.   They need the IL PDBs (in version 1.1), and I don't know who those get published for ASP.NET.

**Comment by JaredShaver** [comment]
@Eilon will the symbols be published for ASP.NET Core 2 by the time it releases?  I was unable to download them from the Microsoft symbol server or the http://srv.symbolsource.org/pdb/Public or http://srv.symbolsource.org/pdb/MyGet servers using ASP.NET Core 2 preview 2.  Being able to step into the source was invaluable when implementing OpenID Connect OWIN middleware and I was hoping I could do the same with ASP.NET Core when implementing the new services/middleware pipeline.

**Comment by Eilon** [comment]
@JaredShaver we're working on it today for ASP.NET Core 2, sorry for the delay!

**Comment by Eilon** [comment]
@JaredShaver the symbols for ASP.NET/EF Core 2.0.0 are being processed, I'll try to update this thread when they're ready.

**Comment by Eilon** [comment]
@JaredShaver - I believe the ASP.NET/EF Core 2.0.0 symbols should be up now. Please give it a try and let us know. (Please note that source links were not enabled in this build, that's something that's still in our plans for future releases.)

**Comment by tiandian** [comment]
I tried with the asp.net core 2.1.0-preview1-26636, but it doesn't work, no symbols found.

**Comment by Eilon** [comment]
Indeed, it appears there might be a problem with the uploaded symbols - we are investigating.

**Comment by JaredShaver** [comment]
@Eilon It looks like the symbols for ASP.NET Core 2 release are loading now, thank you for getting it sorted out.

IdentityModel symbols are still missing though, I'm guessing that is another team's responsibility...  Examples:
- Microsoft.IdentityModel.Protocols.OpenIdConnect.dll
- Microsoft.IdentityModel.Tokens.dll
- System.IdentityModel.Tokens.Jwt.dll
- Microsoft.IdentityModel.Logging.dll
- Microsoft.IdentityModel.Protocols.dll

**Comment by Eilon** [comment]
@brentschmaltz - does your team upload symbols for the IdentityModel packages?

**Comment by brentschmaltz** [comment]
@Eilon we do not.  Can you point us to instructions.

**Comment by Eilon** [comment]
@brentschmaltz I'll send you an email with who to contact.

**Comment by sujayathavale** [comment]
@brentschmaltz, can you confirm if the symbols for the IdentityModel packages are available now from Microsoft Symbol Server?

**Comment by faibistes** [comment]
I can confirm IdentityModel symbols aren't available, and that's biting us hard

**Comment by Legends** [comment]
Just created a breakpoint in my .net core console application (VS2019) at:
`Directory.EnumerateFiles`

Hitting `F11` to step into source code, debug output shows:

> 'dotnet.exe' (CoreCLR: DefaultDomain): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.2.1\System.Private.CoreLib.dll'. Symbols loaded.
> 'dotnet.exe' (CoreCLR: clrhost): Loaded 'C:\Users\Admin\Documents\Visual Studio 2019\Projects\ConsoleAppTest\bin\Debug\netcoreapp2.2\ConsoleAppTest.dll'. Symbols loaded.
> 'dotnet.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.2.1\System.Runtime.dll'. Symbols loaded.
> 'dotnet.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.2.1\System.IO.FileSystem.dll'. Symbols loaded.
> 'dotnet.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.2.1\System.Linq.dll'. Symbols loaded.
> 'dotnet.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.2.1\System.Runtime.Extensions.dll'. Symbols loaded.
> SRCSRV: The module 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.2.1\System.IO.FileSystem.dll' does not contain source server information.

First it says symbols loaded for `FileSystem` and then a the end: The module `FileSystem` does not contain source server information.

So how can I debug .net core code?
